bson/src/test/unit/org/bson/codecs/pojo/PojoCustomTest.java

```diff
@@ -595,6 +597,22 @@ public void testInvalidGetterAndSetterModelDecoding() {
         decodingShouldFail(getCodec(InvalidGetterAndSetterModel.class), "{'integerField': 42, 'stringField': 'myString'}");
     }
 
+    @Test(expected = CodecConfigurationException.class)
+    public void testInvalidBsonRepresentationStringDecoding() {
+        decodingShouldFail(getCodec(BsonRepresentationUnsupportedString.class), "{'id': 'hello', s: 3}");
+    }
+
+    @Test(expected = CodecConfigurationException.class)
+    public void testInvalidBsonRepresentationStringEncoding() {
+        encodesTo(getPojoCodecProviderBuilder(BsonRepresentationUnsupportedString.class),
+                new BsonRepresentationUnsupportedString("1"), "");
+    }
+
+    @Test(expected = CodecConfigurationException.class)
+    public void testInvalidBsonRepresentationIntDecoding() {
+        decodingShouldFail(getCodec(BsonRepresentationUnsupportedInt.class), "{'id': 'hello', s: '3'}");
+    }
```

Comment:
I think you can add a failure test for an invalid BsonRepresentationModel instance as well - where the string value cannot be converted to an ObjectId.

Reply:
Done. It's throwing an IllegalArgumentException in this case now. I can easily change that, but I couldn't think of a more appropriate error message. CodecConfigurationException doesn't sound right because the codec could be configured fine, it's just that the data doesn't match it.